patches/src/main/kotlin/app/revanced/patches/primevideo/ads/SkipAdsPatch.kt

```
@@ -22,24 +25,30 @@ val skipAdsPatch = bytecodePatch(
         // Force doTrigger() access to public so we can call it from our extension.
         doTriggerFingerprint.method.accessFlags = AccessFlags.PUBLIC.value;
 
-        val getPlayerIndex = enterServerInsertedAdBreakStateFingerprint.patternMatch!!.startIndex
         enterServerInsertedAdBreakStateFingerprint.method.apply {
             // Get register that stores VideoPlayer:
             //  invoke-virtual ->getPrimaryPlayer()
             //  move-result-object { playerRegister }
-            val playerRegister = getInstruction<OneRegisterInstruction>(getPlayerIndex + 1).registerA
+            val playerIndex = indexOfFirstInstructionOrThrow() {
+                opcode == Opcode.INVOKE_VIRTUAL && getReference<MethodReference>()?.name == "getPrimaryPlayer"
+            }
+            val playerRegister = getInstruction<OneRegisterInstruction>(playerIndex + 1).registerA
 
             // Reuse the params from the original method:
             //  p0 = ServerInsertedAdBreakState
             //  p1 = AdBreakTrigger
             addInstructions(
-                getPlayerIndex + 2,
+                playerIndex + 2,
                 """
                     invoke-static { p0, p1, v$playerRegister }, Lapp/revanced/extension/primevideo/ads/SkipAdsPatch;->enterServerInsertedAdBreakState(Lcom/amazon/avod/media/ads/internal/state/ServerInsertedAdBreakState;Lcom/amazon/avod/media/ads/internal/state/AdBreakTrigger;Lcom/amazon/avod/media/playback/VideoPlayer;)V
                     return-void
                 """
             )
         }
+
+        // Return early from these callbacks to prevent unwanted overlays from ad breaks.
+        onSeekBehindUnwatchedAdFingerprint.methodOrNull?.returnEarly()
+        onSeekPastUnwatchedAdFingerprint.methodOrNull?.returnEarly()
```

Can be regular method and not orNull. Better to throw an exception if the expected code isn't found.

If this breaks the old target then can drop it. 

Because there's so much un-obfuscated code that's fingerprinted by method/class name, the patch can be pinned to the version you have verified as working. Can bump versions later if someone verifies everything works, so there's no "app doesn't patch / app patches but doesn't work....what version last worked?...someone make a PR to constrain version to that"